js/dydx.js

```diff
@@ -0,0 +1,470 @@
+        //         baselinePositionSize: '9',
+        //         assetResolution: '10000000000',
+        //         syntheticAssetId: '0x4254432d3130000000000000000000'
+        //       },
+        //       ...
+        //    }
+        // }
+        //
+        const result = [];
+        const markets = this.safeValue (marketsResponse, 'markets');
+        const keys = Object.keys (markets);
+        const settleId = 'USDC'; // https://docs.dydx.exchange/#margin
+        for (let i = 0; i < keys.length; i++) {
+            const marketKey = keys[i];
+            const market = markets[marketKey];
+            const id = this.safeString (market, 'market', marketKey);
+            const active = this.safeString (market, 'status') === 'ONLINE';
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const settle = this.safeCurrencyCode (settleId);
+            const symbol = this.symbolDefine (base, quote, settle);
+            const stepSize = this.safeNumber (market, 'stepSize');
+            const tickSize = this.safeNumber (market, 'tickSize');
+            const minOrderSize = this.safeNumber (market, 'minOrderSize');
+            const type = this.safeString (market, 'type');
+            const spot = false;
+            const future = false;
+            const swap = (type === 'PERPETUAL');
+            const option = false;
+            const contract = future || swap;
+            const entry = {
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'settle': settle,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': settleId,
+                'type': type,
+                'spot': spot,
+                'margin': false,
+                'future': future,
+                'swap': swap,
+                'option': option,
+                'active': active,
+                'contract': contract,
+                'linear': undefined,
+                'inverse': undefined,
+                // 'maker': maker, // TODO: I suppose this should be set from base, depending on global market/taker values
+                // 'taker': taker, // TODO: I suppose this should be set from base, depending on global market/taker values
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'price': tickSize,
+                    'amount': stepSize,
+                    // 'base': ..., // TODO: I also think these can be set from base, according to IDs from currencies (whenever possible)
```

Comment:
We don't need these.

Reply:
yes, i've removed now. Just I had copied that from binance class, and maybe there it is odd also. idk.